src/sonic-bgpcfgd/tests/test_bbr.py

```diff
@@ -0,0 +1,319 @@
+        },
+        "dynamic": {
+            "123": {
+                "PEER_V8": ["ipv10", "ipv20"],
+            }
+        }
+    }
+    read_pgs_common(constants, None, expected_bbr_entries)
+
+@patch('bgpcfgd.managers_bbr.log_err')
+def __set_validation_common(key, data, expected_log_err, expected_result, mocked_log_err):
+    cfg_mgr = MagicMock()
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr': cfg_mgr,
+        'tf': TemplateFabric(),
+        'constants': global_constants,
+    }
+    m = BBRMgr(common_objs, "CONFIG_DB", "BGP_BBR")
+    res = m._BBRMgr__set_validation(key, data)
+    assert res == expected_result
+    if expected_log_err is not None:
+        mocked_log_err.assert_called_with(expected_log_err)
+
+def test___set_validation_1():
+    __set_validation_common("all1", {}, "Invalid key 'all1' for table 'BGP_BBR'. Only key value 'all' is supported", False)
+
+def test___set_validation_2():
+    __set_validation_common("all", {"stat": "enabled"}, "Invalid value '{'stat': 'enabled'}' for table 'BGP_BBR', key 'all'. Key 'status' in data is expected", False)
+
+def test___set_validation_3():
+    __set_validation_common("all", {"status": "enabled1"}, "Invalid value '{'status': 'enabled1'}' for table 'BGP_BBR', key 'all'. Only 'enabled' and 'disabled' are supported", False)
+
+def test___set_validation_4():
+    __set_validation_common("all", {"status": "enabled"}, None, True)
+
+def test___set_validation_5():
+    __set_validation_common("all", {"status": "disabled"}, None, True)
+
+def __set_prepare_config_common(status, bbr_enabled_pgs, expected_cmds):
+    cfg_mgr = MagicMock()
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr': cfg_mgr,
+        'tf': TemplateFabric(),
+        'constants': global_constants,
+    }
+    m = BBRMgr(common_objs, "CONFIG_DB", "BGP_BBR")
+    m.directory.data = {"CONFIG_DB__DEVICE_METADATA":
+        {
+            "localhost": {
+                "bgp_asn": "65500"
+            }
+        }
+    }
+    m.bbr_enabled_pgs = bbr_enabled_pgs
+    cmds = m._BBRMgr__set_prepare_config(status)
+    assert cmds == expected_cmds
+
+def test___set_prepare_config_enabled():
+    __set_prepare_config_common("enabled", {
+                "PEER_V4": ["ipv4", "ipv6"],
+                "PEER_V6": ["ipv6"],
+        }, [
+        'router bgp 65500',
+        ' address-family ipv4',
+        '  neighbor PEER_V4 allowas-in 1',
+        ' address-family ipv6',
+        '  neighbor PEER_V4 allowas-in 1',
+        '  neighbor PEER_V6 allowas-in 1',
+    ])
+
+def test___set_prepare_config_disabled():
+    __set_prepare_config_common("disabled", {
+                "PEER_V4": ["ipv4", "ipv6"],
+                "PEER_V6": ["ipv6"],
+        }, [
+        'router bgp 65500',
+        ' address-family ipv4',
+        '  no neighbor PEER_V4 allowas-in 1',
+        ' address-family ipv6',
+        '  no neighbor PEER_V4 allowas-in 1',
```

Comment:
I am a little bit confused here, why do we have PEER_V4 in address-family ipv6?

Reply:
Sorry for misleading here, I added this just for a test:
As you can see from the [source data](https://github.com/Azure/sonic-buildimage/pull/5626/files/671c915dc07e2f270c13b1c80fd55525edfaec81), PEER_V4 works for both ipv4 and ipv6